src/modules/cache-client/auth.ts

```diff
@@ -0,0 +1,50 @@
+import { AxiosInstance, AxiosResponse } from 'axios';
+import { SiweMessage } from 'siwe';
+import { cacheConfigs } from '../../config';
+import { SignerService } from '../signer';
+import { AuthTokens, SiweOptions } from './cache-client.types';
+
+/**
+ * Provides authentication methods to ssi-hub
+ *
+ * @todo Move here authentication parts of `CacheClient`
+ * @todo Make this class top-level module in order to use in application
+ */
+export class SsiAuth {
+  private readonly config: SiweOptions;
+
+  constructor(
+    private signerService: SignerService,
+    private http: AxiosInstance
+  ) {
+    this.config = cacheConfigs()[this.signerService.chainId].auth;
+  }
+
+  /**
+   * Signs in to ssi-hub. Authentication method configurations are looked for in `cacheConfigs`
+   *
+   * @returns authentication tokens
+   */
+  async signIn(): Promise<AxiosResponse<AuthTokens>> {
+    const {
+      data: { nonce },
+    } = await this.http.post<{ nonce: string }>('/login/siwe/initiate');
+    const siweMessage = new SiweMessage({
+      statement: `I accept the Terms of Service: ${this.config.domain}`,
```

Comment:
I would remove the `statement` for now, unless it is necessary.
```suggestion
```

Reply:
Ok